### PR TITLE
feat: Expose new `ApiEndpoints` class to return API endpoint URL

### PR DESCRIPTION
**Pull request by felipecsl**
---
labels: mergeable
---
[//]: #  (Link to the issue corresponding to this chunk of work)
Fixes: #__issue__

## Motivation and Context
This change extracts the API URL building portion out of `FetchHttpClient` into a new class called `ApiEndpoints` which is now exported and allows `js-client-sdk` to then expose the `UFC_ENDPOINT` full URL.

## Description
This is a straightforward refactoring with some decoupling and moving stuff out of `FetchHttpClient`. No functionality should have been changed. No API changes either other than exposing a new class.

Bumped minor version due to new external API being introduced.

## How has this been tested?
Existing and new tests

[//]: # (OPTIONAL)
[//]: # (Add one or multiple labels: enhancement, refactoring, bugfix)

### CODE REVIEW

File: src/client/eppo-client.ts
Comment (felipecsl) [inline src/client/eppo-client.ts:226]: this is a simple/straightforward cleanup for readability. no functionality was changed other than introducing `ApiEndpoints` below

**Review by sameerank — APPROVED** [review]
Looks reasonable to me! Please wait for one more review before merging

File: src/api-endpoints.ts
Comment (sameerank) [reply src/api-endpoints.ts:6]: Or consider a rename to `ConfigEndpoint`? I imagine we would use a different class if we need APIs for non-config purposes in the future

Comment (felipecsl) [reply src/api-endpoints.ts:6]: I made it plural because I see this adding support for other endpoints in the future, not just UFC. I don't feel strongly about it, it's a naming convention that I've seen before and like, but I don't feel strongly about it

**Comment by felipecsl** [comment]
As discussed on Slack, since exposing the API URL to external users my not be needed, I've removed the `ApiEndpoints` export from `index.ts` but the rest of the refactor is still 👍🏽

File: src/client/eppo-client.ts
Comment (leoromanovsky) [inline src/client/eppo-client.ts:224]: this is a cool pattern during reconstruction, thanks

**Review by leoromanovsky — APPROVED** [review]

**Comment by leoromanovsky** [comment]
The test failures are because Sven added new test files (https://github.com/Eppo-exp/sdk-test-data/pull/22) not related to your changes.

Here is an example of my fixes in the swift sdk: https://github.com/Eppo-exp/eppo-ios-sdk/pull/28

**Comment by felipecsl** [comment]
cool I'll merge then and ignore the test failures

**Pull request merged**
